src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlCommand.cs

```diff
@@ -93,7 +84,31 @@ protected override void Clear()
 
             protected override void AfterCleared(SqlCommand owner)
             {
-            
+                owner?.SetCachedCommandExecuteNonQueryAsyncContext(this);
+            }
+        }
+
+        internal sealed class ExecuteXmlReaderAsyncCallContext : AAsyncCallContext<SqlCommand, XmlReader, CancellationTokenRegistration>
+        {
+            public Guid OperationID;
+
+            public SqlCommand Command => _owner;
+            public TaskCompletionSource<XmlReader> TaskCompletionSource => _source;
+
+            public void Set(SqlCommand command, TaskCompletionSource<XmlReader> source, CancellationTokenRegistration disposable, Guid operationID)
```

Comment:
Also consider passing token directly and make it public so it can be used by end method to perform cancellation in parser (since token cannot be retrieved from Registration in netstandard2.0).

Reply:
If we intend to use the token like that then we'll have to make sure that every async operation is written to use a manually written async context. I don't have a problem with that but it might take some time to get that implemented everywhere.